pkg/notes/notes_data.go

```diff
@@ -513,3 +532,74 @@ func setEmptyComponentGitHubRepositoryFromOtherKilnfile(k1, k2 cargo.Kilnfile) c
 }
 
 func closeAndIgnoreError(c io.Closer) { _ = c.Close() }
+
+type TrainstatFetcher interface {
+	FetchTrainstatNotes(milestone string, version string, tile string) ([]string, error)
+}
+
+type TrainstatClient struct {
+	host       string
+	httpClient *http.Client
+}
+
+func NewTrainstatClient(host string) TrainstatClient {
+	if host == "null" || strings.TrimSpace(host) == "" {
+		host = TrainstatUrl()
+	}
+	client := &http.Client{
+		Timeout: time.Second * 10,
+	}
+	return TrainstatClient{
+		host:       host,
+		httpClient: client,
+	}
+}
+
+func (t *TrainstatClient) FetchTrainstatNotes(milestone string, version string, tile string) (notes []string, err error) {
```

Comment:
FetchTrainstatNotes should receive a context as the first parameter and pass it to http.NewRequestWithContext (instead of http.NewRequest).

Reply:
Done!